benchmark_runner/common/grafana/update_grafana_last_value_mappings.py

```diff
@@ -0,0 +1,117 @@
+
+        display_versions = ['ocp_version', 'cnv_version', 'kata_version', 'kata_rpm_version', 'odf_version']
+        for id in ids:
+            data = self.elasticsearch.get_elasticsearch_index_by_id(index='ci-status', id=id)
+            for version, value in data['_source'].items():
+                if version in display_versions and value not in new_versions.values():
+                    new_versions[value.replace(".", "").replace("-", "")] = value
+
+        return new_versions
+
+    def get_value_mappings(self):
+        """
+        This method gets value mappings from main.libsonnet
+        :return:
+        """
+        # Pattern to check in each line
+        start = "        stateTimeline.valueMapping.RegexMap.withOptions("
+        end = "+ stateTimeline.valueMapping.RegexMap.withType('value')"
+
+        # Initialize a flag to indicate whether the lines are within the desired range
+        within_range = False
+        mapping_lines = []
+
+        # Read the content of the file and store lines between start and end patterns in a list
+        with open(self.main_libsonnet_path, 'r') as file:
+            for line in file:
+                if start in line:
+                    within_range = True
+                    continue  # Skip storing the start line
+                elif end in line:
+                    within_range = False
+                    continue  # Skip storing the end line
+                if within_range:
+                    mapping_lines.append(line.strip())
+
+        # Join the lines to create the final string - remove last char )
+        content_string = ''.join(mapping_lines[:-1])
+
+        # Convert the string into a dictionary using json.loads()
+        return json.loads(content_string)
+
+    def update_value_mappings_last_versions(self, last_versions: dict):
+        """
+        This method updated value mapping with last versions
+        :param last_versions:
+        :return:
+        """
+        # Sort the dictionary by "index" values and convert it to a list of tuples
+        sorted_mapping = sorted(self.value_mappings.items(), key=lambda x: x[1]['index'])
+        # Get the maximum index value and the corresponding key
+        max_index, max_key = sorted_mapping[-1][1]['index'], sorted_mapping[-1][0]
+
+        # Use enumerate to add new entries to the value_mapping dictionary
+        for num, (key, value) in enumerate(last_versions.items(), start=1):
+            if not self.value_mappings.get(key):
+                self.value_mappings[key] = {"index": int(max_index) + num, "text": value}
+
+    def update_main_libsonnet(self):
+        """
+        This method updated last versions in main.libsonnet
+        :return:
+        """
+        # Pattern to check in each line
+        start = "        stateTimeline.valueMapping.RegexMap.withOptions("
+        end = "+ stateTimeline.valueMapping.RegexMap.withType('value')"
```

Comment:
Again, find a more robust way to do this.

Reply:
Fixed, see above